aikau/src/main/resources/alfresco/services/CloudSyncService.js

```diff
@@ -0,0 +1,568 @@
+       * This will be called regardless of whether or not the current user is authenticated (as even if the
+       * user is not authenticated then an HTTP status code of 200 will be returned). If the response indicates
+       * that the user is not authenticated then a dialog will be shown to allow the user to authenticate.
+       * 
+       * @instance
+       * @param {object} response The response from the request
+       * @param {object} originalRequestConfig The configuration passed on the original request
+       */
+      onCloudAuthenticationReponse: function alfresco_services_CloudSyncService__onCloudAuthenticationReponse(response, originalRequestConfig) {
+         var known = lang.getObject("known", false, response);
+         if (known === false) 
+         {
+            // Publish authentication dialog
+            this.alfServicePublish(topics.CREATE_FORM_DIALOG, {
+               dialogId: "ALF_CLOUD_AUTHENTICATION_DIALOG",
+               dialogTitle: "cloud-auth.dialog.title",
+               dialogCloseTopic: topics.CLOUD_AUTHENTICATION_SUCCESS,
+               formSubmissionTopic: topics.CLOUD_AUTHENTICATION_REQUEST,
+               formSubmissionGlobal: true,
+               formSubmissionPayloadMixin: originalRequestConfig.data,
+               showValidationErrorsImmediately: false,
+               widgets: this.widgetsForAuthenticationDialog
+            });
+         }
+         else
+         {
+            // Show a dialog to select the location to sync to
+            this.showCloudLocationPicker(originalRequestConfig.data);
+         }
+      },
+
+      /**
+       * This is the failure callback for the XHR request made from the 
+       * [onCloudSyncRequest]{@link module:alfresco/services/CloudSyncService#onCloudSyncRequest} function.
+       * It will only be called when the request fails (e.g. because the server cannot be reached) rather than
+       * the user not being authenticated.
+       * 
+       * @instance
+       * @param {object} response The response from the request
+       * @param {object} originalRequestConfig The configuration passed on the original request
+       * @fires module:alfresco/core/topics#DISPLAY_PROMPT
+       */
+      onCloudAuthenticationReponseFailure: function alfresco_services_CloudSyncService__onCloudAuthenticationReponseFailure(/*jshint unused:false*/ response, originalRequestConfig) {
+         this.alfServicePublish(topics.DISPLAY_PROMPT, {
+            message: this.message("cloud-auth.connection.message")
+         });
+      },
+
+      /**
+       * Publishes a request to show a form [dialog]{@link module:alfresco/dialogs/AlfDialog} containing the 
+       * the [widgetsForSyncDialog]{@link module:alfresco/services/CloudSyncService#widgetsForSyncDialog}
+       * model.
+       * 
+       * @instance
+       * @param {object} payload The payload containing the nodes requested to be sync'd
```

Comment:
It's always called with a payload... it just happens that this is not the function bound to the subscription.